code/_onclick/hud/screen_objects.dm

```diff
@@ -268,12 +278,24 @@
 							selecting = "eyes"
 
 	if(old_selecting != selecting)
-		update_icon()
+		update_icon(usr)
 	return 1
 
-/obj/screen/zone_sel/update_icon()
+/obj/screen/zone_sel/update_icon(mob/user)
 	overlays.Cut()
 	overlays += image('icons/mob/screen_gen.dmi', "[selecting]")
+	user.zone_selected = selecting
+
+/obj/screen/zone_sel/alien
+	icon = 'icons/mob/screen_alien.dmi'
+
+/obj/screen/zone_sel/alien/update_icon(mob/user)
+	overlays.Cut()
+	overlays += image('icons/mob/screen_alien.dmi', "[selecting]")
+	user.zone_selected = selecting

```

Comment:
this fixes the xeno's zone selection using the wrong overlays, the ones in screen_alien.dmi are black, not red. And selecting the head no longer wrongly shows an alien head.